iptables/iptables.go

```diff
@@ -304,15 +305,21 @@ func Exists(table Table, chain string, rule ...string) bool {
 
 	// try -C
 	// if exit status is 0 then return true, the rule exists
-	if _, err := Raw(append([]string{
-		"-t", string(table), "-C", chain}, rule...)...); err == nil {
+	_, err := Raw(append([]string{"-t", string(table), "-C", chain}, rule...)...)
+	if err == nil {
 		return true
 	}
+	if !strings.Contains(err.Error(), "unknown option") {
+		return false
+	}
 
 	// parse "iptables -S" for the rule (this checks rules in a specific chain
 	// in a specific table)
-	ruleString := strings.Join(rule, " ")
-	ruleString = chain + " " + ruleString
+	return existsRaw(table, chain, rule...)
+}
+
+func existsRaw(table Table, chain string, rule ...string) bool {
+	ruleString := fmt.Sprintf("%s %s\n", chain, strings.Join(rule, " "))
```

Comment:
Looking for the `\n` is in fact to avoid a possible match on a rule subset. 

The existing raw exist check does nothing but looking for the rule string in the output of `iptables -t <table> -S <chain>`, which is a list of `\n` terminated strings,  where each rule follows this format:

`-<Action> <Chain> <rule args>\n`

Given the rule format, checking if the byte stream contains `<rule>\n` will guarantee no subset can match.